kedro-airflow/kedro_airflow/grouping.py

```diff
@@ -113,3 +113,37 @@ def dfs(cur_node_name: str, component: int) -> None:
                 group_dependencies[new_name_parent].append(new_name_child)
 
     return group_to_seq, group_dependencies
+
+
+def group_by_namespace(
+    pipeline: Pipeline,
+) -> tuple[dict[str, list[Node]], dict[str, list[str]]]:
+    """
+    Groups nodes based on their namespace using Pipeline's grouped_nodes_by_namespace property.
+    Non-namespaced nodes are assigned to a default namespace.
+    """
+    nodes_by_namespace: dict[str, list[Node]] = {}
+    dependencies_by_namespace: dict[str, list[str]] = {}
+
+    grouped_nodes = pipeline.grouped_nodes_by_namespace
+
+    node_to_namespace = {}
+
+    for group_name, group_info in grouped_nodes.items():
+        ns = group_info["name"] if group_info["type"] == "namespace" else "__default__"
```

Comment:
I don't think you need to set a default. Ideally it should be -
- group the namespace 
- individual nodes per task if not namespaced

Which is how the `grouped_by_namespace` property `dict` also returns to. I think the issue with the default would be that it could create a somewhat circular dependencies with the execution. For example when pipeline1 is not namespaced i.e. is default, pipeline2 is namespaced and pipeline3 is not namespaced, the execution order would be impossible to determine

Reply:
That's a question that I had yeah, it didn't follow the behavior of the results that come from the grouped_nodes_by_namespace function but it made sense when I was writing it.